heudiconv/parser.py

```diff
@@ -60,58 +62,87 @@ def find_files(regex, topdir=op.curdir, exclude=None,
             yield path
 
 
-def get_extracted_dicoms(fl):
-    """Given a list of files, possibly extract some from tarballs.
+def get_extracted_dicoms(
+        fl: typing.Iterable[str]
+        ) -> typing.ItemsView[typing.Optional[int], typing.List[str]]:
+    """Given a collection of files and/or directories, possibly extract 
+    some from tarballs.
 
-    For 'classical' heudiconv, if multiple tarballs are provided, they correspond
-    to different sessions, so here we would group into sessions and return
-    pairs  `sessionid`, `files`  with `sessionid` being None if no "sessions"
-    detected for that file or there was just a single tarball in the list
-    """
-    # TODO: bring check back?
-    # if any(not tarfile.is_tarfile(i) for i in fl):
-    #     raise ValueError("some but not all input files are tar files")
+    Parameters
+    ----------
+    fl : Iterable[str]
```

Comment:
I am not sure how valuable (I dislike duplication) is type annotation here.   I would say strip them off.
Unfortunately, apparently this question is not yet addressed within numpy itself , so there is no definitive mandate/answer - https://github.com/numpy/numpydoc/issues/196 , so I would have gone for 

```suggestion
    fl
```

Reply:
in that issue I was pointed out to https://pypi.org/project/sphinx-autodoc-typehints/ , which seems might work with numpy style , could you check it out?

Reply:
Yeah, can do. Were you thinking about this as a separate issue, or something to figure out for this pull?

Reply:
sphinx extension -- separate PR probably. Here -- I would just keep it concise aiming for future sanity of the tools and the extension ;)

Reply:
Hmm, it seems like that extension might already be (partly) integrated.

https://github.com/nipy/heudiconv/blob/c83e06c6d3c71a43c3d570f9c05d12f36dc3a066/docs/conf.py#L44

Though, not sure whether another configuration needs to be set

Reply:
FWIW, the extension seems to more or less "just work" by adding `sphinx_autodoc_typehints` to 

https://github.com/nipy/heudiconv/blob/c83e06c6d3c71a43c3d570f9c05d12f36dc3a066/docs/conf.py#L43-L4

And the extension also handles return types from the signature.